src/util/compareThought.js

```diff
@@ -43,7 +43,7 @@ export const compareLowercase = (a, b) => compare(lower(a), lower(b))
 export const comparePunctuationAndOther = (a, b) => {
   const aIsPunctuation = regexPunctuation.test(a)
   const bIsPunctuation = regexPunctuation.test(b)
-  return aIsPunctuation && !bIsPunctuation ? -1
+  return (aIsPunctuation && !bIsPunctuation) || (a === '') ? -1
```

Comment:
I would prefer this in its own comparator function to keep things clean. I'm very proud of my comparator composition 😂. Please define `compareEmpty` and add it to the `makeOrderedComparator` argument list.

Reply:
Sure.